advisor/src/main/kotlin/Advisor.kt

```diff
@@ -113,3 +119,37 @@ class Advisor(
             AdvisorRun(startTime, endTime, Environment(), config, results)
         }
 }
+
+fun AdvisorResult.normalizeVulnerabilityData(): AdvisorResult =
+    copy(vulnerabilities = vulnerabilities.normalizeVulnerabilityData())
+
+fun List<Vulnerability>.normalizeVulnerabilityData(): List<Vulnerability> =
+    map { vulnerability ->
+        val normalizedReferences = vulnerability.references.map { reference ->
+            reference
+                .let {
+                    // Treat "MODERATE" as an alias for "MEDIUM" independently of the scoring system.
+                    if (it.severity == "MODERATE") reference.copy(severity = "MEDIUM") else it
```

Comment:
Should be `it.copy`, but maybe also switch from `let` to `run` altogether.

Reply:
BTW, this is one of the cases where I've wondered whether we should make use of `var` instead of `val` in model data classes. Being able to reassign properties instead of making copies of the parent object would be so much more memory-friendly, and probably also result in more readable code.

Reply:
So you want to do `pandora.box.open = true` instead of `pandora.copy(box = pandora.box.copy(open = true))`? 😜

Reply:
Yes, for some selected properties in selected data classes where we *expect* data to be fixed-up. If we had proper abstractions of the data models from the APIs, this could be made so that such fixups are only allowed by the producers of the data, but not the consumers (which get an immutable view on the data).